jupyterhub/tests/test_roles.py

```diff
@@ -1046,3 +884,34 @@ async def test_oauth_allowed_roles(app, create_temp_role):
     app_service = app.services[0]
     assert app_service['name'] == 'oas1'
     assert set(app_service['oauth_roles']) == set(allowed_roles)
+
+
+async def test_config_role_assignment():
+    role_name = 'painter'
+    user_name = 'benny'
+    other_users = ['agnetha', 'bjorn', 'anni-frid']
+    roles_to_load = [
+        {
+            'name': role_name,
+            'description': 'painting with colors',
+            'scopes': ['users', 'groups'],
+            'users': other_users,
+        },
+    ]
+    for user_in_config in [False, True]:
+        if user_in_config:
+            roles_to_load[0]['users'].append(user_name)
+        kwargs = {'load_roles': roles_to_load}
+        hub = MockHub(**kwargs, clean_db=False)
+        hub.init_db()
```

Comment:
I think init_db always starts with an empty database in MockHub, so I'm not sure if it's testing multiple initializations from one database. Try:

1. initialize and load roles
2. check presence
3. change `.load_roles` and re-run only `init_role_assignment()`
4. check membership again (it should change)

Reply:
Thanks, rewrote the test